### PR TITLE
Add replicate `ContributeAndFinalize` statuses and transitions

### PR DESCRIPTION
**Pull request by mcornaton**
- [ ] My code has been formatted as explained in [CONTRIBUTING.md](https://github.com/iExecBlockchainComputing/documentation/blob/v8-staging/CONTRIBUTING.md)

### CODE REVIEW

File: for-developers/advanced/task-feedback.md [hunk 1]
```diff
@@ -143,6 +143,9 @@ While the _task_ holds a meta status, each _replicate_ has its own status which
 | `RESULT_UPLOADING` | The worker is uploading the result |
 | `RESULT_UPLOAD_FAILED` | The upload of the result failed |
 | `RESULT_UPLOADED` | The result has been uploaded to IPFS over the _iExec Result Proxy_ (standard or TEE _tasks_) or to Dropbox (TEE only), dependending on the _deal_ parameters |
+| `CONTRIBUTE_AND_FINALIZE_ONGOING` | The worker sent the "contributeAndFinalize(...)" transaction on chain |
+| `CONTRIBUTE_AND_FINALIZE_DONE` | The worker has contributed and finalized the task. The last is now considered as completed on-chain |
```
Comment (jbern0rd) [inline for-developers/advanced/task-feedback.md]:
```suggestion
| `CONTRIBUTE_AND_FINALIZE_DONE` | The worker has contributed and finalized the task. The task is now considered as completed on-chain |
```

Comment (jbern0rd) [reply for-developers/advanced/task-feedback.md]: or 
```suggestion
| `CONTRIBUTE_AND_FINALIZE_DONE` | The worker has contributed and finalized the task. The latter is now considered as completed on-chain |
```

File: for-developers/advanced/task-feedback.md [hunk 2]
```diff
@@ -151,7 +154,7 @@ While the _task_ holds a meta status, each _replicate_ has its own status which
 | `RECOVERING` | The worker has been stopped, it is starting back from where it stopped |
 | `WORKER_LOST` | The worker didn't ping the iexec-core scheduler for a while. It is considered as out for this _task_ |
 
-The transitions between those states are as follows:
+The transitions between those states are as follows. Please note that all failed status - pictured in red in the following diagram - finally lead to `ABORTED`. For the sake of simplicity, this final status has not been represented here.
```
Comment (jbern0rd) [inline for-developers/advanced/task-feedback.md]: Should we render this important note with an `hint` ?

File: for-developers/advanced/task-feedback.md [hunk 3]
```diff
@@ -206,29 +208,29 @@ subgraph Result upload stage
...
-REVEAL_FAILED --> ABORTED
-RESULT_UPLOAD_FAILED --> ABORTED
-COMPLETE_FAILED --> ABORTED
-
 %% Style definitions
 classDef failure fill:#a00
 classDef completed fill:#0a0
 ```
 
+ContributeAndFinalize mode is a particular case that makes the _tasks_ execution faster and cheaper. In that mode, the _workers_ can contribute and finalize the tasks themselves. As the _workers_ have to be trustworthy to enable that mode, it is required the apps run in Trusted Execution Environments. Another limitation iExec strives to remove is the non-callback mode.
```
Comment (jbern0rd) [inline for-developers/advanced/task-feedback.md]: Really interesting precision. I find it is written too late in the document.
Should we write it earlier and maybe invert the diagram and the table ?

File: for-developers/advanced/task-feedback.md (hunk 1 above)
Comment (mcornaton) [reply for-developers/advanced/task-feedback.md]: The latter suits me better :grin:

File: for-developers/advanced/task-feedback.md (hunk 2 above)
Comment (mcornaton) [reply for-developers/advanced/task-feedback.md]: Sure. Please see https://github.com/iExecBlockchainComputing/documentation/pull/171/commits/201722e6ecc0d86cae448b9f6e558602b61f407a.

File: for-developers/advanced/task-feedback.md (hunk 3 above)
Comment (mcornaton) [reply for-developers/advanced/task-feedback.md]: I have reordered those elements. Please see https://github.com/iExecBlockchainComputing/documentation/pull/171/commits/201722e6ecc0d86cae448b9f6e558602b61f407a.

File: for-developers/advanced/task-feedback.md [hunk 4]
```diff
@@ -116,42 +116,19 @@ task1
...
-| `FAILED` | The worker failed to participate to the _task_ |
-| `ABORTED` | The scheduler asked the worker to stop working on this _replicate_ while the latter was still working on it |
-| `RECOVERING` | The worker has been stopped, it is starting back from where it stopped |
-| `WORKER_LOST` | The worker didn't ping the iexec-core scheduler for a while. It is considered as out for this _task_ |
+See the following flowchart for details on their transitions.
 
-The transitions between those states are as follows:
+{% hint style="warning" %}
+
+Please ote that all failed status - pictured in red in the following diagram - finally lead to `ABORTED`. For the sake of simplicity, this final status has not been represented here.
```
Comment (jbern0rd) [inline for-developers/advanced/task-feedback.md]:
```suggestion
Please note that all failed status - pictured in red in the following diagram - finally lead to `ABORTED`. For the sake of simplicity, this final status has not been represented here.
```

Comment (jbern0rd) [reply for-developers/advanced/task-feedback.md]: Should we write ?
```suggestion
Please note that all failed status - pictured in red in the following diagram - finally lead to the `ABORTED` status. For the sake of simplicity, this final status has not been represented here.
```

File: for-developers/advanced/task-feedback.md [hunk 5]
```diff
@@ -185,11 +162,10 @@ subgraph Compute stage
 end
 
 subgraph Contribute stage
-    COMPUTED --> CONTRIBUTING
+    COMPUTED -- If trust != 1\nor standard task\nor contribution have already been made\nor callback mode --> CONTRIBUTING
```
Comment (jbern0rd) [inline for-developers/advanced/task-feedback.md]: There is a mix of singular and plural here : `contribution have already been made`
Should it be `at least one contribution has already been made` ?

File: for-developers/advanced/task-feedback.md [hunk 6]
```diff
@@ -206,29 +182,65 @@ subgraph Result upload stage
     RESULT_UPLOADING --> RESULT_UPLOAD_FAILED:::failure
 end
 
+subgraph Contribute and Finalize stage
+    COMPUTED -- If trust = 1\nand TEE task\nand no contribution have been made yet\nand not in callback mode --> CONTRIBUTE_AND_FINALIZE_ONGOING
```
Comment (jbern0rd) [inline for-developers/advanced/task-feedback.md]: Same remark between singular and plural here

File: for-developers/advanced/task-feedback.md (hunk 4 above)
Comment (mcornaton) [reply for-developers/advanced/task-feedback.md]: I don't think we need the precision. `ABORTED` seems clear enough for me. But let's wait for other points of view.

File: for-developers/advanced/task-feedback.md (hunk 5 above)
Comment (mcornaton) [reply for-developers/advanced/task-feedback.md]: Please see https://github.com/iExecBlockchainComputing/documentation/pull/171/commits/984f2d516fdaae28ae059e71623cb26dce6d544a.

File: for-developers/advanced/task-feedback.md (hunk 6 above)
Comment (mcornaton) [reply for-developers/advanced/task-feedback.md]: Please see https://github.com/iExecBlockchainComputing/documentation/pull/171/commits/984f2d516fdaae28ae059e71623cb26dce6d544a.

File: for-developers/advanced/task-feedback.md [hunk 7]
```diff
@@ -116,42 +116,19 @@ task1
 └── replicate3 (workerZ)
 ```
 
-While the _task_ holds a meta status, each _replicate_ has its own status which can be one of these:
+A replicate status workflow can follow two different flows:
+1. Usual flow: the _task_ is replicated on a number of _workers_, depending on required trust. The _Scheduler_ has to notify the _workers_ when the consensus is reached. It should also finalize the _task_ on-chain.
```
Comment (jbern0rd) [inline for-developers/advanced/task-feedback.md]: Should it be ?
```suggestion
1. Contribute Reveal Finalize flow: the _task_ is replicated on a number of _workers_, depending on required trust. The _Scheduler_ has to notify the _workers_ when the consensus is reached. It should also finalize the _task_ on-chain.
```

File: for-developers/advanced/task-feedback.md [hunk 8]
```diff
@@ -116,42 +116,19 @@ task1
...
+2. ContributeAndFinalize flow: the _task_ is finalized by the only _worker_ that has worked on it. It makes the _task_ execution faster and cheaper. However, some limitations apply:
```
Comment (jbern0rd) [inline for-developers/advanced/task-feedback.md]:
```suggestion
2. Contribute And Finalize flow: the _task_ is finalized by the only _worker_ that has worked on it. It makes the _task_ execution faster and cheaper. However, some limitations apply:
```

File: for-developers/advanced/task-feedback.md (hunk 7 above)
Comment (jbern0rd) [reply for-developers/advanced/task-feedback.md]: It is to raise a discussion. We may want to ask this question to more people.

File: for-developers/advanced/task-feedback.md (hunk 4 above)
Comment (jbern0rd) [reply for-developers/advanced/task-feedback.md]: No problem for me. This is already quite good as it is.

File: for-developers/advanced/task-feedback.md (hunk 3 above)
Comment (jbern0rd) [reply for-developers/advanced/task-feedback.md]: Thanks :pray:

File: for-developers/advanced/task-feedback.md (hunk 2 above)
Comment (jbern0rd) [reply for-developers/advanced/task-feedback.md]: Thanks :pray:

**Review by jbern0rd — APPROVED** [review]

File: for-developers/advanced/task-feedback.md (hunk 8 above)
Comment (ericr6) [reply for-developers/advanced/task-feedback.md]: is "Contribute and Finalize" too complex to introduce the 2 paths?
a suggestion below
1: usual flow (default): ....
2: optimized flow (running under conditions):...

Comment (mcornaton) [reply for-developers/advanced/task-feedback.md]: Thanks for the idea! Here it is: https://github.com/iExecBlockchainComputing/documentation/pull/171/commits/bf822a0428a100b98cb97bb9326ef48a6714142f.

**Pull request merged**